src/mixins/name-request-mixin.ts

```diff
@@ -57,13 +59,15 @@ export default class NameRequestMixin extends Mixins(DateMixin) {
    * @param nr the name request response payload
    * */
   isNrValid (nr: any): boolean {
+    let requestActionCDList = [NameRequestTypes.CHANGE_OF_NAME, NameRequestTypes.CONVERSION]
+    if (this.isEntityTypeFirm) requestActionCDList.pop()
```

Comment:
So now this mixin needs to know about entity type and has a special case for firms.

It would be better if the acceptable NR types were stored in the resource files (and if they were passed in to this method).

Reply:
This change now looks good. 👍